db/dbformat.cc

```diff
@@ -131,7 +131,7 @@ void InternalKeyComparator::FindShortestSeparator(
   Slice user_limit = ExtractUserKey(limit);
   std::string tmp(user_start.data(), user_start.size());
   user_comparator_->FindShortestSeparator(&tmp, user_limit);
-  if (tmp.size() < user_start.size() &&
+  if (tmp.size() <= user_start.size() &&
```

Comment:
Changing `<` to `<=` seems like a bigger change than changing `<=` to nothing. Also, it's unlikely that any existing implementations of `FindShortestSeparator()` can make key longer.

If user key has increased, it's safe to change the sequence number to anything. Length of the user key is irrelevant for correctness, afaict. I guess, it was here as an optimization to avoid calling comparator if `FindShortestSeparator()` didn't make any changes. Before this PR there wouldn't be any use in having `FindShortestSeparator()` change key without shortening it. With this PR it can even make sense for user-provided `FindShortestSeparator()` to lengthen the key in some weird cases, e.g. if the separated keys are "a" and "b", and the user is going to use `iterate_upper_bound = "a9", it would make sense for `FindShortestSeparator()` to return "a9" rather than "a".